test/check-machines-create

```diff
@@ -331,6 +331,9 @@ class TestMachinesCreate(VirtualMachinesCase):
                                                       storage_size=128, storage_size_unit='MiB',
                                                       start_vm=True,
                                                       delete=False))
+        # Wait for virt-install to define the VM and then stop it
+        wait(lambda: "pxe-guest" in self.machine.execute("virsh list --persistent"), delay=3)
+        b.wait_in_text("#vm-pxe-guest-state", "Running")
         self.machine.execute("virsh destroy pxe-guest")
```

Comment:
Is there actually any specific reason why we create the VM with `start_vm=True`, and then we just immediately destroy it? Instead of running createTest with start_vm=False?

Reply:
I suppose not -- I can try with `start_vm=False`. I get some local failures, happy to look into it. Would you be okay with doing this as a follow-up? (GitHub is broken, and this test blocks other PRs)

Reply:
I tried this:

```diff
--- test/check-machines-create
+++ test/check-machines-create
@@ -333,12 +333,10 @@ class TestMachinesCreate(VirtualMachinesCase):
                                                       name="pxe-guest",
                                                       location=f"type=direct,source={iface}",
                                                       storage_size=128, storage_size_unit='MiB',
-                                                      start_vm=True,
+                                                      start_vm=False,
                                                       delete=False))
-        # Wait for virt-install to define the VM and then stop it
-        wait(lambda: "pxe-guest" in self.machine.execute("virsh list --persistent"), delay=3)
-        b.wait_in_text("#vm-pxe-guest-state", "Running")
-        self.machine.execute("virsh destroy pxe-guest")
+        # Wait for virt-install to define the VM
+        wait(lambda: "pxe-guest" in self.machine.execute("virsh list --name --all --persistent"), delay=3)
 
         # Verify that the newly created disk is first in the boot order and the network used for the PXE boot is not on the bootable devices list
         b.wait_in_text("#vm-pxe-guest-boot-order", "disk")
```

But that fails -- the boot order is still "network, disk". So starting the VM does something important to it. As this is more than "trivial cleanup", I'll leave it at that for now -- of course, please feel free to have a look.